docs/docs/environment-variables.md

```diff
@@ -2,34 +2,76 @@
 title: "Environment Variables"
 ---
 
+## Environments and Environment Variables
+
 You can easily provide environment variables to your site.
 
-For JavaScript loaded into the browser, just add a `.env.development` and/or
-`.env.production` file in your root folder for development or production builds
-respectively. The environment variables are embedded during build time using
-Webpack's [DefinePlugin](https://webpack.js.org/plugins/define-plugin/). Because
-these variables are provided at build time, you will need to restart your dev
-server or rebuild your site after changing them.
+Note that we need to distinguish in this discussion between variables which are been defined in
+special places in order to be used in different deployment environments, and true OS-level 
+environment variables that could be used in, for example, command-line calls.
+We'll call the former "Project Environment Variables" and the latter "OS Environment Variables".
+In both cases we want to be able to access the relavant value of these variable for the environment
+we're in.
+
+By default gatsby supports only 2 environments:
+
+ * If you run `gatsby develop`, then you will be in the 'development' environment.
+ * If you run `gatsby build` + `gatsby serve`, then you will be in the 'production' environment.
+ 
+If you want to define other Environments then you'll need to do a little more work. See 'Additional Environments' below.
+
+## Accessing Environment Variables in JavaScript
+
+All of the Project and OS Environment Variables are only directly available at build time, or
+when Node.Js is running. They aren't immediately available at run time of the client code; they
+need to be actively captured and embedded into our client-side JavaScript.
+This is achieved during the build using Webpack's [DefinePlugin](https://webpack.js.org/plugins/define-plugin/).
+
+Once the Environment Variables have been embedded into the client-side, they are accessible from the
```

Comment:
Here I've gone with _environment variables_ when not specifically referring to _OS Env Vars_ or _Project Env Vars_

```suggestion
Once the environment variables have been embedded into the client-side, they are accessible from the
```